### ISSUE TITLE
Not getting some Infobox data in some languages

### ISSUE THREAD

**Issue by siznax** [siznax/wptools#91]
Someone pointed this out [on Stack Overflow](https://stackoverflow.com/questions/34189426/parse-wikipedia-wikitext-template-named-parameters-to-extract-data-from-taxobox/39480770#39480770). In particular, getting some infobox data in French does not work. Supposedly, this is because Infoboxes have a different format in different languages?

```python
>>> page = wptools.page('Okapi', lang='fr')
>>> page.get_parse()
fr.wikipedia.org (parse) Okapi
Okapi (fr) data
{
  infobox: <dict(0)>
...
```

[French](https://fr.wikipedia.org/wiki/Okapi?action=raw) wikitext:

```
{{Taxobox début | animal | ''Okapia johnstoni'' | Okapi2.jpg | Okapi }}
{{Taxobox | embranchement | Chordata }}
{{Taxobox | classe | Mammalia }}
{{Taxobox | sous-classe | Theria }}
{{Taxobox | ordre | Artiodactyla }}
{{Taxobox | famille | Giraffidae }}
{{Taxobox taxon | animal | genre | Okapia | [[Edwin Ray Lankester|Lankester]], [[1901]] }}
{{Taxobox taxon | animal | espèce | Okapia johnstoni | ([[Philip Lutley Sclater|Sclater]], [[1901]]) }}
{{Taxobox synonymes |
* ''Equus johnstoni'' <small>P.L. Sclater, 1901</small>}}
{{Taxobox UICN | EN | A2abcd+4abcd }}
{{Taxobox répartition | Okapi map.jpg }}
{{Taxobox répartition | Okapi distribution.PNG }}
{{Taxobox fin}}
```

The same article in **German** works fine:

```python
>>> page = wptools.page('Okapi', lang='de')
>>> page.get_parse()
de.wikipedia.org (parse) Okapi
Okapi (de) data
{
  infobox: <dict(21)> Taxon3_Name, Bild, Taxon2_Rang, Taxon5_Rang,...
...
```

[German](https://de.wikipedia.org/wiki/Okapi?action=raw) wikitext:

```
{{Taxobox
| Taxon_Name       = Okapi
| Taxon_WissName   = Okapia johnstoni
| Taxon_Rang       = Art
| Taxon_Autor      = ([[Philip Lutley Sclater|P. L. Sclater]], 1901)
| Taxon2_WissName  = Okapia
| Taxon2_Rang      = Gattung
| Taxon2_Autor     = [[Ray Lankester|Lankester]], 1901
| Taxon3_Name      = Giraffenartige
| Taxon3_WissName  = Giraffidae
| Taxon3_Rang      = Familie
| Taxon4_Name      = Stirnwaffenträger
| Taxon4_WissName  = Pecora
| Taxon4_Rang      = ohne Rang
| Taxon5_Name      = Wiederkäuer
| Taxon5_WissName  = Ruminantia
| Taxon5_Rang      = Unterordnung
| Taxon6_Name      = Paarhufer
| Taxon6_WissName  = Artiodactyla
| Taxon6_Rang      = Ordnung
| Bild             = Okapi2.jpg
| Bildbeschreibung = 
}}
```

Likewise in **English**:

```python
>>> page = wptools.page('Okapi', lang='en')
>>> page.get_parse()
en.wikipedia.org (parse) Okapi
en.wikipedia.org (imageinfo) File:Okapi2.jpg
Okapi (en) data
{
  infobox: <dict(11)> status, range_map, status_ref, name, range_m...
...
```

[English](https://en.wikipedia.org/wiki/Okapi?action=raw) wikitext:

```
{{Speciesbox
| name = Okapi
| image = Okapi2.jpg
| image_caption = An okapi at [[Disney's Animal Kingdom]] in [[Florida]].
| status = EN
| status_system = IUCN3.1
| status_ref = <ref name=iucn>{{IUCN2008|assessor=IUCN SSC Antelope Specialist Group|year=2008|id=15188|title=Okapia johnstoni|downloaded=26 November 2013}} Database entry includes a brief justification of why this species is endangered.</ref>
| parent_authority = [[Ray Lankester|Lankester]], 1901
| taxon = Okapia johnstoni
| authority = ([[P.L. Sclater]], 1901)
}}
```

And **Italian**:

```python
>>> page = wptools.page('Okapi', lang='it')
>>> page.get_parse()
it.wikipedia.org (parse) Okapi
Okapia johnstoni (it) data
{
  infobox: <dict(44)> classe, regno, sottoclasse, sinonimi, didasc...
...
```

[Italian]() wikitext:

```
{{Tassobox
|nome=Okapi
|statocons=EN
|statocons_versione=iucn3.1
|statocons_ref=<ref name=iucn>{{IUCN|summ=15188|autore=Mallon, D., Kümpel, N., Quinn, A., Shurter, S., Lukas, J., Hart, J. & Mapilanga, J. 2013}}</ref>
|immagine=Okapi2.jpg
|didascalia=Un okapi al [[Disney's Animal Kingdom]]
<!-- CLASSIFICAZIONE: -->
|dominio=[[Eukaryota]]
|regno=[[Animalia]]
|sottoregno=
|superphylum=
|phylum=[[Chordata]]
|subphylum=
|infraphylum=
|microphylum=
|nanophylum=
|superclasse=
|classe=[[Mammalia]]
|sottoclasse=
|infraclasse=
|superordine=
|ordine=[[Artiodactyla]]
|sottordine=
|infraordine=
|superfamiglia=
|famiglia=[[Giraffidae]]
|sottofamiglia=
|tribù=
|sottotribù=
|genere='''Okapia'''
|genereautore=[[Edwin Ray Lankester|Lankester]], [[1901]]
|sottogenere=
|specie='''O. johnstoni'''
|sottospecie=
<!-- NOMENCLATURA BINOMIALE: -->
|biautore=([[Philip Lutley Sclater|P. L. Sclater]]
|binome=Okapia johnstoni
|bidata=[[1901]])
<!-- NOMENCLATURA TRINOMIALE: -->
|triautore=
|trinome=
|tridata=
<!-- ALTRO: -->
|sinonimi?=
|sinonimi=
|nomicomuni=
|suddivisione=
|suddivisione_testo=
}}
```

By comparison, the French Speciesbox syntax looks mal-formed.

**Comment by siznax** [comment]
I'm not sure if we'll ever solve this for everyone. We should focus on Wikidata instead.

**Comment by siznax** [comment]
Initial comparison of parsetrees.

English parsetree snippet:

```
>>> en_okapi = wptools.page('Okapi', lang='en')
>>> en_okapi.get_parse()
>>> print en_okapi.data['parsetree']
...
<template lineStart="1"><title>Speciesbox
</title><part><name> name </name><equals>=</equals><value> Okapi
</value></part><part><name> image </name><equals>=</equals><value> Okapi2.jpg
</value></part><part><name> image_caption </name><equals>=</equals><value> An okapi at [[Disney's Animal Kingdom]] in [[Florida]].
</value></part><part><name> status </name><equals>=</equals><value> EN
</value></part><part><name> status_system </name><equals>=</equals><value> IUCN3.1
</value></part><part><name> status_ref </name><equals>=</equals><value> <ext><name>ref</name><attr> name=iucn</attr><inner>{{IUCN2008|assessor=IUCN SSC Antelope Specialist Group|year=2008|id=15188|title=Okapia johnstoni|downloaded=26 November 2013}} Database entry includes a brief justification of why this species is endangered.</inner><close>&lt;/ref&gt;</close></ext>
</value></part><part><name> parent_authority </name><equals>=</equals><value> [[Ray Lankester|Lankester]], 1901
</value></part><part><name> taxon </name><equals>=</equals><value> Okapia johnstoni
</value></part><part><name> authority </name><equals>=</equals><value> ([[P.L. Sclater]], 1901)
</value></part></template>
...
```

French parsetree snippet:

```python
>>> fr_okapi = wptools.page('Okapi', lang='fr')
>>> fr_okapi.get_parse()
>>> print fr_okapi.data['parsetree']
...
<template lineStart="1"><title>Taxobox début </title><part><name index="1"/><value> animal </value></part><part><name index="2"/><value> ''Okapia johnstoni'' </value></part><part><name index="3"/><value> Okapi2.jpg </value></part><part><name index="4"/><value> Okapi </value></part></template>
<template lineStart="1"><title>Taxobox </title><part><name index="1"/><value> embranchement </value></part><part><name index="2"/><value> Chordata </value></part></template>
<template lineStart="1"><title>Taxobox </title><part><name index="1"/><value> classe </value></part><part><name index="2"/><value> Mammalia </value></part></template>
<template lineStart="1"><title>Taxobox </title><part><name index="1"/><value> sous-classe </value></part><part><name index="2"/><value> Theria </value></part></template>
<template lineStart="1"><title>Taxobox </title><part><name index="1"/><value> ordre </value></part><part><name index="2"/><value> Artiodactyla </value></part></template>
<template lineStart="1"><title>Taxobox </title><part><name index="1"/><value> famille </value></part><part><name index="2"/><value> Giraffidae </value></part></template>
<template lineStart="1"><title>Taxobox taxon </title><part><name index="1"/><value> animal </value></part><part><name index="2"/><value> genre </value></part><part><name index="3"/><value> Okapia </value></part><part><name index="4"/><value> [[Edwin Ray Lankester|Lankester]], [[1901]] </value></part></template>
<template lineStart="1"><title>Taxobox taxon </title><part><name index="1"/><value> animal </value></part><part><name index="2"/><value> espèce </value></part><part><name index="3"/><value> Okapia johnstoni </value></part><part><name index="4"/><value> ([[Philip Lutley Sclater|Sclater]], [[1901]]) </value></part></template>
<template lineStart="1"><title>Taxobox synonymes </title><part><name index="1"/><value>
* ''Equus johnstoni'' &lt;small&gt;P.L. Sclater, 1901&lt;/small&gt;</value></part></template>
<template lineStart="1"><title>Taxobox UICN </title><part><name index="1"/><value> EN </value></part><part><name index="2"/><value> A2abcd+4abcd </value></part></template>
<template lineStart="1"><title>Taxobox répartition </title><part><name index="1"/><value> Okapi map.jpg </value></part></template>
<template lineStart="1"><title>Taxobox répartition </title><part><name index="1"/><value> Okapi distribution.PNG </value></part></template>
<template lineStart="1"><title>Taxobox fin</title></template>
...
```

**Comment by siznax** [comment]
Btw, someone was nice enough to point this out [on Stack Overflow](https://stackoverflow.com/questions/34189426/parse-wikipedia-wikitext-template-named-parameters-to-extract-data-from-taxobox/39480770#39480770).

**Comment by siznax** [comment]
Getting Infobox data for some French articles does work:

```python
>>> page = wptools.page('Napoleon', lang='fr')
>>> page.get_parse()
fr.wikipedia.org (parse) Napoleon
fr.wikipedia.org (imageinfo) File:Jacques-Louis David - The Emper...
Napoléon Ier (fr) data
{
  infobox: <dict(65)> image, profession, date de décès, conjoint, ...
...
```

[French](https://fr.wikipedia.org/wiki/Napol%C3%A9on_Ier?action=raw) wikitext:

```
{{Infobox Personnalité politique
 | nom                           = Napoléon {{Ier}}
 | charte                        = Monarque
 | image                         = Jacques-Louis David - The Emperor Napoleon in His Study at the Tuileries - Google Art Project 2.jpg
 | légende                       = ''[[Napoléon dans son cabinet de travail]]'', [[Jacques-Louis David]], 1812.
 | dynastie                      = [[Famille Bonaparte|Maison Bonaparte]]
 | nom de naissance              = Napoleone Bonaparte <small>(sur l'acte de baptême)</small>
 | date de naissance             = {{date de naissance|15 août 1769}}
 | date de décès                 = {{Date de décès|5|mai|1821|15|août|1769}}
 | lieu de naissance             = [[Ajaccio]] ([[Royaume de France|France]])
 | lieu de décès                 = [[Sainte-Hélène (île)|Île Sainte-Hélène]] ([[Royaume-Uni de Grande-Bretagne et d'Irlande|Royaume-Uni]])
 | nature du décès               = 
 | sépulture                     = 
 | nationalité                   = 
 | parti = 
| père                           = [[Charles Bonaparte]]
 | mère                          = [[Maria Letizia Ramolino]]
 | fratrie                       = 
 | conjoint                      = [[Joséphine de Beauharnais]] <small>(1796-1809)</small><br />[[Marie-Louise d'Autriche]] <small>(1810-1821)</small>
 | enfants                       = [[Napoléon II|Napoléon Bonaparte]], ''prince impérial'', ''roi de Rome''
 | héritier                      = Prince [[Napoléon II|Napoléon]], ''prince impérial''
 | entourage                     = 
 | profession                    = 
 | religion                      = [[Catholicisme romain]]
 | résidence                     = 
 | signature                     = Firma Napoleón Bonaparte.svg 
 | emblème                       = Imperial Coat of Arms of France (1804-1815).svg
 | emblème2                      = Coat of Arms of the Kingdom of Italy (1805-1814).svg
 | liste                         = [[Liste des monarques de France|Monarques de France]]·[[Liste des rois d'Italie|Monarques d'Italie]]
| fonction1                      = [[Liste des monarques de France|Empereur des Français]]
 | depuis le fonction1 = 
| à partir du fonction1          = {{date|18 mai 1804}}
 | jusqu'au fonction1            = {{date|6 avril 1814}} <br /><small>({{Durée|18|5|1804|6|4|1814}})</small>
 | couronnement 1                = {{date|2 décembre 1804}},<br /> en la [[cathédrale Notre-Dame de Paris]]
 | prédécesseur 1                = ''Lui-même'' <small>([[Consulat (histoire de France)|Premier consul de la République]])</small>
 | successeur 1                  = {{souverain2|Louis XVIII}}<small>(roi de France)</small>
 | à partir du fonction2         = {{date|20 mars 1815-}}
 | jusqu'au fonction2            = {{date|22 juin 1815}}<br /><small>({{Durée|20|3|1815|22|6|1815}})</small>
 | prédécesseur 2                = {{souverain2|Louis XVIII}}<small>(roi de France)</small>
 | successeur 2                  = {{Napoléon II}} <small>(prétendant)</small><br />{{souverain2|Louis XVIII}} <small>(roi de France)</small>
 | fonction3                     = [[Liste des rois d'Italie|Roi d'Italie]]
 | à partir du fonction3         = {{date|17 mars 1805}}
 | jusqu'au fonction3            = {{date|11 avril 1814}} <br /><small>({{Durée|17|3|1805|11|4|1814}})</small>
 | prédécesseur 3                = ''Lui-même'' <small>(président de la République italienne)</small>
 | successeur 3                  = {{souverain2|Victor-Emmanuel II}} <br /><small>(roi d'Italie en 1861)</small>
 | fonction4                     = [[Liste des souverains de Germanie|Protecteur de la Confédération du Rhin]]
 | à partir du fonction4         = {{date|12 juillet 1806}}
 | jusqu'au fonction4            = {{date|19 octobre 1813}} <br /><small>({{Durée|12|7|1806|19|10|1813}})</small>
 | prédécesseur 4                = {{souverain2|François Ier d'Autriche|François II}} <br /><small>(empereur romain germanique)''</small>
 | successeur 4                  = [[Confédération germanique]]
 | fonction5                     = [[Acte de Médiation|Médiateur de la Confédération suisse]]
 | à partir du fonction5         = {{date|19 février 1803}}
 | jusqu'au fonction5            = {{date|19 octobre 1813}} <br /><small>({{Durée|19|2|1803|19|10|1813}})</small>
 | prédécesseur 5                = [[République helvétique]]
 | successeur 5                  = [[Confédération des XXII cantons|Confédération des {{nobr|{{XXII}} cantons}}]]
 | fonction6                     = [[République italienne (1802-1805)|Président de la République italienne]]<br />''Napoléon Bonaparte''
 | à partir du fonction6         = {{date|26 janvier 1802}}
 | jusqu'au fonction6            = {{date|17 mars 1805}} <br /><small>({{Durée|26|1|1802|17|3|1805}})</small>
 | vice-président 6              = [[Francesco Melzi d'Eril]]
 | prédécesseur 6                = [[République cisalpine]]
 | successeur 6                  = ''Lui-même'' <small>(roi d'Italie)</small>
 | fonction7                     = [[Consulat (histoire de France)|Premier consul de la République]]<br />''Napoléon Bonaparte''
 | à partir du fonction7         = {{date|10 novembre 1799}}
 | jusqu'au fonction7            = {{date|18 mai 1804}} <br /><small>({{Durée|10|11|1799|18|04|1804}})</small>
 | prédécesseur 7                = [[Directoire]]
 | successeur 7                  = ''Lui-même'' <small>(empereur des Français)</small>
 }}
```

**Comment by siznax** [comment]
Not getting the **Spanish** Infobox data for this example either, because the name of the template does not include "box". 

```python
>>> page = wptools.page('Okapi', lang='es')
>>> page.get_parse()
es.wikipedia.org (parse) Okapi
es.wikipedia.org (parse) Okapi
Okapia johnstoni (es) data
{
  iwlinks: <list(3)> https://commons.wikimedia.org/wiki/Okapia_joh...
  pageid: 34357
  parsetree: <str(14076)> <root><template><title>Ficha de taxón</t...
  requests: <list(1)> parse
  title: Okapia johnstoni
  wikibase: Q82037
  wikidata_url: https://www.wikidata.org/wiki/Q82037
  wikitext: <str(11752)> {{Ficha de taxón| name = Okapi| status = ...
}
```

We could improve things by adding an argument for "infobox token".

**Comment by siznax** [comment]
I added the `boxterm` parameter to help capture Infoboxes that do not have "box" in the title. The previous example failed because we were looking for "box" in the title: "Ficha de taxón".  By adding `boxterm=Ficha` we can now get the Infobox data:

```python
>>> page = wptools.page('Okapi', lang='es', boxterm='Ficha')
>>> page.get_parse()
es.wikipedia.org (parse) Okapi
es.wikipedia.org (imageinfo) File:Okapi2.jpg
Okapia johnstoni (es) data
{
  image: <list(1)> {u'size': 628222, 'kind': 'parse-image', u'desc...
  infobox: <dict(19)> status, range_map, binomial_authority, trend...
  iwlinks: <list(3)> https://commons.wikimedia.org/wiki/Okapia_joh...
  pageid: 34357
  parsetree: <str(14070)> <root><template><title>Ficha de taxón</t...
  requests: <list(2)> parse, imageinfo
  title: Okapia johnstoni
  wikibase: Q82037
  wikidata_url: https://www.wikidata.org/wiki/Q82037
  wikitext: <str(11746)> {{Ficha de taxón| name = Okapi| status = ...
}
```

```python
>>> page.data['infobox']
{'binomial': "''Okapia johnstoni''",
 'binomial_authority': '([[Philip Sclater|Sclater]], [[1901]])',
 'classis': '[[Mammalia]]',
 'familia': '[[Giraffidae]]',
 'genus': "'''''Okapia'''''",
 'genus_authority': '[[Ray Lankester|Lankester]], 1901',
 'image': 'Okapi2.jpg',
 'image_width': '250px',
 'name': 'Okapi',
 'ordo': '[[Artiodactyla]]',
 'phylum': '[[Chordata]]',
 'range_map': 'Okapi_distribution.PNG',
 'range_map_caption': u'Distribuci\xf3n del okapi',
 'regnum': '[[Animalia]]',
 'species': "'''''O. johnstoni'''''",
 'status': 'EN',
 'status_ref': '<ext><name>ref</name><attr> name=iucn</attr><inner>{{IUCN |asesores=Mallon, D., K&#252;mpel, N., Quinn, A., Shurter, S., Lukas, J., Hart, J. &amp; Mapilanga, J.|a&#241;o=2013|edici&#243;n IUCN=2013.1|consultado=17 de septiembre de 2013}}</inner><close>&lt;/ref&gt;</close></ext>',
 'status_system': 'iucn3.1',
 'trend': 'stable'}
```

**Comment by siznax** [comment]
I've added a more general infobox parser alternative that should handle "unusual" infobox wikitext:

```python
>>> page = wptools.page('Okapi', lang='fr')
>>> page.get_parse()
fr.wikipedia.org (parse) Okapi
Okapi (fr) data
{
  infobox: <dict(2)> count, boxes
  ...
}
```

```python
>>> page.data['infobox']['count']
13

>>> page.data['infobox']['boxes']
[{u'Taxobox d\xe9but': [[{'index': '1'}, 'animal'],
   [{'index': '2'}, "''Okapia johnstoni''"],
   [{'index': '3'}, 'Okapi2.jpg'],
   [{'index': '4'}, 'Okapi']]},
 {'Taxobox': [[{'index': '1'}, 'embranchement'],
   [{'index': '2'}, 'Chordata']]},
 {'Taxobox': [[{'index': '1'}, 'classe'], [{'index': '2'}, 'Mammalia']]},
 {'Taxobox': [[{'index': '1'}, 'sous-classe'], [{'index': '2'}, 'Theria']]},
 {'Taxobox': [[{'index': '1'}, 'ordre'], [{'index': '2'}, 'Artiodactyla']]},
 {'Taxobox': [[{'index': '1'}, 'famille'], [{'index': '2'}, 'Giraffidae']]},
 {'Taxobox taxon': [[{'index': '1'}, 'animal'],
   [{'index': '2'}, 'genre'],
   [{'index': '3'}, 'Okapia'],
   [{'index': '4'}, '[[Edwin Ray Lankester|Lankester]], [[1901]]']]},
 {'Taxobox taxon': [[{'index': '1'}, 'animal'],
   [{'index': '2'}, u'esp\xe8ce'],
   [{'index': '3'}, 'Okapia johnstoni'],
   [{'index': '4'}, '([[Philip Lutley Sclater|Sclater]], [[1901]])']]},
 {'Taxobox synonymes': [[{'index': '1'},
    "* ''Equus johnstoni'' <small>P.L. Sclater, 1901</small>"]]},
 {'Taxobox UICN': [[{'index': '1'}, 'EN'], [{'index': '2'}, 'A2abcd+4abcd']]},
 {u'Taxobox r\xe9partition': [[{'index': '1'}, 'Okapi map.jpg']]},
 {u'Taxobox r\xe9partition': [[{'index': '1'}, 'Okapi distribution.PNG']]},
 {'Taxobox fin': []}]
```

**Comment by siznax** [comment]
This has been fixed with release [v0.4.10](https://github.com/siznax/wptools/releases/tag/v0.4.10).

**Comment by AmoghM** [comment]
I found that the infobox output for Hindi doesn't yield result in the same language.

The details mention for a page [here ](https://hi.wikipedia.org/wiki/%E0%A4%A8%E0%A4%BE%E0%A4%93%E0%A4%AE%E0%A5%80_%E0%A4%93%E0%A4%B8%E0%A4%BE%E0%A4%95%E0%A4%BE)in the infobox are in Hindi. However, when hitting it yields us the keys of the infobox dictionary in English.

```
>>> page = wptools.page('नाओमी ओसाका', lang='hi')
>>> page.get_parse()
```
```
hi.wikipedia.org (parse) नाओमी ओसाका
नाओमी ओसाका (hi) data
{
  image: <list(0)> 
  infobox: <dict(34)> name, image, caption, country, residence, bi...
  pageid: 1100542
  parsetree: <str(13947)> <root><template><title>Infobox tennis bi...
  requests: <list(1)> parse
  title: नाओमी ओसाका
  wikibase: Q17466583
  wikidata_url: https://www.wikidata.org/wiki/Q17466583
  wikitext: <str(9736)> {{Infobox tennis biography| name          ...
}
```


```
>>> print(page.data['infobox']) #the keys should also be in Hindi but we get in English such as 'name' and 'country' etc.
```

```
{{'name': 'नाओमी ओसाका',
 'image': '[[File:Osaka WMQ15 (6) (19957109051).jpg|250px]]',
 'caption': '[[2015 विंबलडन चैंपियनशिप]] में नाओमी',
 'country': '{{JPN}}',
 'residence': '[[बोका रातोंन]] फ्लोरिडा, संयुक्त राज्य',
 'birth_date': '{{birth date and age|1997|10|16|mf|=|yes}}',
 'birth_place': 'चुओ-कू, [[ओसाका]], जापान',
...
```
It will be great to know your view on this output and figure out a fix, if possible.
